core/trino-main/src/main/java/io/trino/sql/planner/plan/MergeProcessorNode.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.plan;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import io.trino.sql.planner.Symbol;
+
+import java.util.List;
+
+import static io.trino.sql.planner.plan.TableWriterNode.MergeTarget;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A node that contains the Searched CASE and RIGHT JOIN derived from a
```

Comment:
`and RIGHT JOIN derived from a` could you elaborate more in a javadoc what it means?

Is the node essentially a partitioned, distributed join where right side contains rows to be merged?

Reply:
I think the "contains" part is confusing. It should say "processes the result of"